script/check-config.sh

```diff
@@ -43,6 +55,8 @@ is_set_as_module() {
 }
 
 color() {
+	[ -v NO_COLOR ] && return
```

Comment:
nit: In theory, according to [the spec](https://no-color.org/) this should be

```suggestion
	[ -n "$NO_COLOR" ] && return
```

(Because `NO_COLOR=` should be treated as `NO_COLOR` **not** being set.)

But it doesn't really matter.

Reply:
This should be either 

```bash
[ -v NO_COLOR ] && [ -n "$NO_COLOR" ] && return`
```

or

```bash
[ -n "${NO_COLOR:-}" ] && return
```

(otherwise it will trip `set -u`).

Reply:
BTW [no-color.org](https://no-color.org) lists conforming implementations; out of curiosity I looked at random 3 of them ([bashkit](https://bashkit.wuage.io/), [go-ansi](https://git.sr.ht/~mpldr/go-ansi/tree/master/item/no-color.go) and [color-debug](https://github.com/roboticslab-uc3m/color-debug)) -- all three merely check if the variable exists. The fourth one ([termenv](https://github.com/muesli/termenv)) actually checks if the variable contents is not empty.

Anyway, changed to `[ -n "${NO_COLOR:-}" ] && return`.

PTAL @cyphar

Reply:
ping @cyphar

Reply:
Yup, sounds good.

@kolyshkin That might be the case, but the text of their spec explicitly says that `NO_COLOR=` *must not* disable colours.